lib/ddtrace/initialization.rb

```diff
@@ -0,0 +1,59 @@
+module Datadog
+  # Responsible for the behavior the tracer after it's completely
+  # loaded, but before `require 'ddtrace'` returns control to
+  # the user.
+  class Initialization
+
+    # @param tracer [Datadog] an application-level Datadog APM tracer object
+    def initialize(tracer)
```

Comment:
For this whole PR, there should exist a separate runtime/initializer for each of our separate products (profiler initializer/runtime, ci initialize/runtime).

I haven't had the chance to refactor it as such, but each product should own its lifecycle separately.

`lib/ddtrace/initialization.rb` will likely bring them all together (or expose precise hooks so that profiler, ci, etc can hook themselves in).